src/main/java/com/datadoghq/connect/logs/sink/DatadogLogsApiWriter.java

```diff
@@ -97,20 +117,29 @@ private JsonArray formatBatch(String topic) {
             }
 
             JsonElement recordJSON = recordToJSON(record);
-            JsonObject message = populateMetadata(topic, recordJSON, record.timestamp());
+            JsonObject message = populateMetadata(topic, recordJSON, record.timestamp(), () -> kafkaHeadersToJsonElement(record));
```

Comment:
Shouldn't we have this change behind a `config.parseRecordHeaders` test?
I'm thinking about the extra toJson/fromJson call even for people that don't want to use these information.

Reply:
I've added a test for the `config.parseRecordHeaders`: https://github.com/DataDog/datadog-kafka-connect-logs/pull/49/files#diff-6cefcd933c4ff34173ccb8e393231ab406748e7d05f8f1d3c29221f66ce551ebR201-R202

Also, I've added the parameter as a supplier so that `kafkaHeadersToJsonElement` is not executed if `config.parseRecordHeaders` is not  set to `true`

Reply:
kafkaHeadersToJsonElement method is passed as a Supplier into the populateMetadata. That means that the code is executed when .get() is called, in our case here:
`if (config.parseRecordHeaders) {
            content.add("kafkaheaders", kafkaHeaders.get());
        }`